MetabolicAtlas/data-generation
# feat: integrated gems timeline

This PR is related to https://github.com/MetabolicAtlas/MetabolicAtlas/issues/949.

**Type of change** 

**List of changes made** 
- Add timeline chart generator for integrated GEMs.
- Re-write import and export statements using ES Modules (`import` instead of `require`). This is needed because modern versions of `d3` need ES Modules and do not work with `require`.
- Minor code fixes that now need to be fixed after switching to ES Modules. For example there is a few instances where variables were not declared before they were used.
- Prettier code formatting has been applied to these associated files.

**Testing**
- A certain branch in other repos is needed
- A rebuild is needed due to new dependencies
- Instructions on how to test
  1. This PR requires the work from https://github.com/MetabolicAtlas/data-files/pull/26.
  2. Create the `gemRepository` folder if necessary: `mkdir gemRepository`.
  3. Install new dependencies: `yarn`.
  4. Run the script as usual, for example: `yarn start ../data-files`.
  5. Verify that an SVG file is added to the `gemRepository` folder.
